cinder/client.go

```diff
@@ -0,0 +1,264 @@
+package cinder
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+type TokenFn func() string
+
+func SetEndpointFn(endpoint *url.URL, wrappedHandler RequestHandlerFn) RequestHandlerFn {
+	return func(req *http.Request) (*http.Response, error) {
+		req.URL.Host = endpoint.Host
+		req.Host = endpoint.Host
+		return wrappedHandler(req)
+	}
+}
+
+func SetAuthHeaderFn(token TokenFn, wrappedHandler RequestHandlerFn) RequestHandlerFn {
+	return func(req *http.Request) (*http.Response, error) {
+		req.Header.Set("X-Auth-Token", token())
+		return wrappedHandler(req)
+	}
+}
+
+func NewClient(tenantId string, handleRequest RequestHandlerFn) *Client {
+	return &Client{tenantId, handleRequest}
+}
+
+type Client struct {
+	tenantId      string
+	handleRequest RequestHandlerFn
+}
+
+// Shows information for a specified snapshot.
```

Comment:
These aren't auto-generated, right? Would prefer to stick with conventional "GetSnapshot ..."